app/src/main/java/com/example/sharingang/DetailedItemFragment.kt

```diff
@@ -73,6 +79,62 @@ class DetailedItemFragment : Fragment() {
         return binding.root
     }
 
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        inflater.inflate(R.menu.menu_detailed, menu)
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        super.onPrepareOptionsMenu(menu)
+        val sell = menu.findItem(R.id.menuSell)
+        val resell = menu.findItem(R.id.menuResell)
+        if (!args.item.userId.equals(currentUserProvider.getCurrentUserId())) {
+            menu.findItem(R.id.menuEdit).isVisible = false
+            sell.isVisible = false
+            resell.isVisible = false
+        } else {
+            sell.isVisible = !(item?.sold ?: args.item.sold)
+            resell.isVisible = item?.sold ?: args.item.sold
+        }
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.menuEdit -> {
+                view?.findNavController()?.navigate(
+                    DetailedItemFragmentDirections.actionDetailedItemFragmentToNewEditFragment(args.item)
+                )
+                true
+            }
+            R.id.menuSell -> {
+                updateSold()
+                true
+            }
+            R.id.menuResell -> {
+                updateSold()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
```

Comment:
Good use of when btw! Note that you can put multiple conditions, to trigger the block if any of them is true:
```suggestion
        return when (item.itemId) {
            R.id.menuEdit -> {
                view?.findNavController()?.navigate(
                    DetailedItemFragmentDirections.actionDetailedItemFragmentToNewEditFragment(args.item)
                )
                true
            }
            R.id.menuSell, R.id.menuResell -> {
                updateSold()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
```

Reply:
Oh nice, didn't know about that!